include/fam/fam_c.h

```diff
@@ -0,0 +1,983 @@
+/*
+ * Copyright (c) 2022 Hewlett Packard Enterprise Development, LP. All
+ * rights reserved. 
+ * 
+ * Redistribution and use in source and binary forms, with or
+ * without modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ *    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+ * IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * See https://spdx.org/licenses/BSD-3-Clause
+ *
+ */
+#ifndef FAM_C_H_
+#define FAM_C_H_
+
+#include <stdint.h>
+#include <sys/stat.h>
+#include <fam/fam.h>
+
+#ifndef int128_t
+#ifdef __int128
+typedef __int128 int128_t;
+#else
+#ifdef __SIZEOF_INT128__
+typedef __int128 int128_t;
+#else
+typedef long long int128_t;
+#endif // __SIZEOF_INT128__
+#endif // __int128
+#endif // int128_t
+
+typedef void c_fam;
```

Comment:
I think it is better to use void* instead of void as datatype. 
We expect c_fam and c_fam_context object, region and data item descriptor to be of a pointer type and not just void.

Reply:
We have kept the declarations similar to the C++ datatypes. Hence, while using c_fam and c_fam_context they have to be used as c_fam* and c_fam_context*, just like fam* and fam_context*.